data-prepper-api/src/main/java/org/opensearch/dataprepper/model/sink/AbstractSink.java

```diff
@@ -76,7 +78,7 @@ public void output(Collection<T> records) {
     @Override
     public void shutdown() {
         if (retryThread != null) {
-            retryThread.stop();
```

Comment:
In Java 21, `Thread::stop` always throws an `IllegalArgumentException`. So I've changed this approach to tell the runnable below to stop itself.

Reply:
Do we have other locations in plugins where threads are stopped like this?

Reply:
I ran IntelliJ's "Find Usages" feature on `Thread::stop`. It found this usage, but found no others. So I don't believe so.